pkg/tsdb/azuremonitor/azure-resource-graph-datasource.go

```diff
@@ -0,0 +1,274 @@
+package azuremonitor
+
+import (
+	"bytes"
+	"strings"
+
+	// "compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/api/pluginproxy"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util/errutil"
+	"github.com/opentracing/opentracing-go"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+// AzureResourceGraphDatasource calls the Azure Resource Graph API's
+type AzureResourceGraphDatasource struct {
+	httpClient    *http.Client
+	dsInfo        *models.DataSource
+	pluginManager plugins.Manager
+}
+
+// AzureResourceGraphQuery is the query request that is built from the saved values for
+// from the UI
+type AzureResourceGraphQuery struct {
+	RefID        string
+	ResultFormat string
+	URL          string
+	Model        *simplejson.Json
+	Params       url.Values
+	Target       string
+}
+
+// executeTimeSeriesQuery does the following:
+// 1. build the AzureMonitor url and querystring for each query
+// 2. executes each query by calling the Azure Monitor API
+// 3. parses the responses for each query into the timeseries format
+func (e *AzureResourceGraphDatasource) executeTimeSeriesQuery(ctx context.Context, originalQueries []plugins.DataSubQuery,
+	timeRange plugins.DataTimeRange) (backend.QueryDataResponse, error) {
+	result := backend.QueryDataResponse{
+		Responses: map[string]backend.DataResponse{},
+	}
+
+	queries, err := e.buildQueries(originalQueries, timeRange)
+	if err != nil {
+		return backend.QueryDataResponse{}, err
+	}
+
+	for _, query := range queries {
+		result.Responses[query.RefID] = e.executeQuery(ctx, query, originalQueries, timeRange)
+	}
+
+	return result, nil
+}
+
+func (e *AzureResourceGraphDatasource) buildQueries(queries []plugins.DataSubQuery,
+	timeRange plugins.DataTimeRange) ([]*AzureResourceGraphQuery, error) {
+	azureResourceGraphQueries := []*AzureResourceGraphQuery{}
+
+	for _, query := range queries {
+		queryBytes, err := query.Model.Encode()
+		if err != nil {
+			return nil, fmt.Errorf("failed to re-encode the Azure Resource Graph query into JSON: %w", err)
+		}
+
+		queryJSONModel := argJSONQuery{}
+		err = json.Unmarshal(queryBytes, &queryJSONModel)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode the Azure Resource Graph query object from JSON: %w", err)
+		}
+
+		azureResourceGraphTarget := queryJSONModel.AzureResourceGraph
+		azlog.Debug("AzureResourceGraph", "target", azureResourceGraphTarget)
+
+		resultFormat := azureResourceGraphTarget.ResultFormat
+		if resultFormat == "" {
+			resultFormat = timeSeries
+		}
+
+		params := url.Values{}
+		rawQuery, err := KqlInterpolate(query, timeRange, azureResourceGraphTarget.Query)
```

Comment:
Don't we need a default value here?

Reply:
No we don't. The timestamp filter will be entered by users.